src/ui_utils.c

```diff
@@ -1919,7 +1919,9 @@ GtkWidget *ui_path_box_new(const gchar *title, GtkFileChooserAction action, GtkE
 	vbox = gtk_vbox_new(FALSE, 0);
 	if (gtk_widget_get_parent(path_entry))	/* entry->parent may be a GtkComboBoxEntry */
 	{
-		GtkWidget *parent = gtk_widget_get_parent(path_entry);
+		GtkWidget *parent = path_entry, *next_parent;
+		while ((next_parent = gtk_widget_get_parent(parent)) != NULL)
+			parent = next_parent;
```

Comment:
while at it doing something like that, we could simplify the code by always resolving the parent and using that, dropping the conditional here -- e.g. doing what this block does, but unconditionally.  If the first `get_parent()` returns NULL, it's the same as the else block anyway.